### PR TITLE
adapted links to documentation to new website layout

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java [hunk 1]
```diff
@@ -107,11 +107,11 @@ private Extension getExtension(Feature feature) {
         String version = feature.getVersion();
         String link = null;
         if (type.equals("binding")) {
-            link = "http://docs.openhab.org/addons/bindings/" + name + "/readme.html";
+            link = "https://www.openhab.org/addons/bindings/" + name;
```
Comment (ghys) [inline bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java]: You can avoid a unnecessary redirect by adding a trailing slash.

File: bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java [hunk 2]
```diff
@@ -107,11 +107,11 @@ private Extension getExtension(Feature feature) {
...
         } else if (type.equals("action")) {
-            link = "http://docs.openhab.org/addons/actions/" + name + "/readme.html";
+            link = "https://www.openhab.org/addons/actions/" + name;
```
Comment (ghys) [inline bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java]: Same here.

File: bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java [hunk 3]
```diff
@@ -107,11 +107,11 @@ private Extension getExtension(Feature feature) {
...
         } else if (type.equals("persistence")) {
-            link = "http://docs.openhab.org/addons/persistence/" + name + "/readme.html";
+            link = "https://www.openhab.org/addons/persistence/" + name;
```
Comment (ghys) [inline bundles/org.openhab.core.karaf/src/main/java/org/openhab/core/karaf/internal/KarafExtensionService.java]: And here.

**Comment by kaikreuzer** [comment]
@ghys Better?

**Comment by ghys** [comment]
@kaikreuzer  By the way, https://github.com/search?p=6&q=org%3Aopenhab+docs.openhab.org&type=Code returns quite a lot of links to docs.openhab.org, and I won't even mention the community forum and other places :)
Definitely overlooked this a bit. 😞 
Maybe a banner or something like that in the current Jekyll layout saying the current stable docs have moved to "(new URL -possibly dynamically computed)" would be good?

**Comment by kaikreuzer** [comment]
@ghys Yeah, that's why my original requirement was to keep all existing subdomains and urls in place as they are - changing urls is always a pain :-(

Yes, I think we should add a banner on Friday to it - could you prepare something like that?
As we have to keep the old pages for the time being, we'll probably also have to update the footer links for the impress and privacy policy...

**Comment by ghys** [comment]
> Yes, I think we should add a banner on Friday to it - could you prepare something like that?
As we have to keep the old pages for the time being, we'll probably also have to update the footer links for the impress and privacy policy...

Sure.

In fact the idea was to keep using `docs.openhab.org` for the latest snapshot's docs and the archives of previous releases, while the current _stable_ version moved to `openhab.org/docs` and would be the default. Right now it could be a little confusing for the average user likely using the stable version to land by default on the docs for a unreleased snapshot (including new unreleased bindings etc.).

The snapshot docs for the snapshot could then also be migrated to VuePress to match the website's style later - and the old URLs would still work, with a few redirects for moved content as necessary.

**Comment by kaikreuzer** [comment]
> In fact the idea was to keep using docs.openhab.org for the latest snapshot's docs

No, I am very strongly against that - the old homepage is legacy and no one should be forced to look at it any longer. Keeping it available for the old releases and existing (old) links that point there is ok, but I would certainly expect the latest (i.e. snapshot) version to be on the new website.

> Right now it could be a little confusing for the average user likely using the stable version to land by default on the docs for a unreleased snapshot (including new unreleased bindings etc.).

Yes, but this is what we had for many months now and it was indeed only "little" confusing - so I think we can keep it this way until we come up with something better. Wrt add-ons that don't exist on the release, but only in snapshots, would we have a chance to add a marker on their page (or a small notice header)? I'd be fine to maintain a separate list of add-on ids, if necessary.

**Comment by ghys** [comment]
Okay, then I misunderstood what was discussed in the forum thread or assumed wrong, and I actually prefer it like you describe it ;)

> would we have a chance to add a marker on their page (or a small notice header)?

Definitely possible. Not just for add-ons in fact, but you could maintain a list of pages whose content isn't relevant for the current stable version (new major features like units of measurement etc.) and they'd get an auto-included little warning box at the top.
In time it would even be possible to include it inline in existing articles, and even make it auto-disappear when the targeted version is released (just thinking off the top of my head).

**Pull request merged**